sensorhub-service-swe/src/main/java/org/sensorhub/impl/client/sos/SOSClient.java

```diff
@@ -75,28 +79,98 @@
     DataComponent dataDescription;
     DataEncoding dataEncoding;
     boolean useWebsockets;
-    volatile boolean started;
     
+    /**
+     * Background threads that perform asynchronous work for this client. Right now this is only used when streaming
+     * observations to a caller.
+     */
+    ScheduledThreadPoolExecutor backgroundThreads;
     
-    public interface SOSRecordListener
-    {
-        public void newRecord(DataBlock data);
+    /**
+     * Counts the number of times that we have attempted to reconnect to the remote server for streaming observations.
+     * This is set to zero when streaming is started and incremented every time that the connection fails and we try to
+     * reconnect.
+     */
+    int streamingRetryAttempts;
+
+    /**
+     * The maximum number of times that we should attempt to reconnect to the remote server when streaming observations
+     * and the connection fails. If the value is negative, then there is no limit to the number of reconnect attempts.
+     * If the value is zero, do not attempt to reconnect at all.
+     */
+    int maxStreamingRetryAttempts = -1;
```

Comment:
Could we make this configurable by adding a [RobustConnectionConfig](https://github.com/opensensorhub/osh-core/blob/237f16afec6977164d0287936be8f1a7c99a730b/sensorhub-core/src/main/java/org/sensorhub/impl/module/RobustConnectionConfig.java) to [SWEVirtualSensorConfig](https://github.com/opensensorhub/osh-core/blob/237f16afec6977164d0287936be8f1a7c99a730b/sensorhub-service-swe/src/main/java/org/sensorhub/impl/sensor/swe/SWEVirtualSensorConfig.java)?